code/modules/unit_tests/lungs.dm

```diff
@@ -0,0 +1,145 @@
+	if(lab_rat.has_alert(ALERT_TOO_MUCH_CO2) && (!test_lungs.safe_co2_max || !test_mix.has_gas(/datum/gas/carbon_dioxide)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_TOO_MUCH_CO2 when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_TOO_MUCH_PLASMA) && (!test_lungs.safe_plasma_max || !test_mix.has_gas(/datum/gas/plasma)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_TOO_MUCH_PLASMA when it wasn't expeced.")
+
+/// Tests the standard lungs organ to ensure it behaves the way we expect.
+/datum/unit_test/lungs_sanity/Run()
+	// Tests the "standard" form of breathing.
+	// 50 Litres of O2/N2 gas mix, ideal for life.
+	var/datum/gas_mixture/test_mix = create_standard_mix()
+	var/obj/item/organ/internal/lungs/test_lungs = allocate(/obj/item/organ/internal/lungs)
+	var/mob/living/carbon/human/lab_rat = allocate(/mob/living/carbon/human/consistent)
+	// Test one breath of O2/N2 mix.
+	var/breath_ok = test_lungs.check_breath(test_mix, lab_rat)
+	// Return status code 0 indicates failure to respirate.
+	LUNGS_TEST_RETURN(breath_ok, test_lungs)
+	LUNGS_TEST_BREATHE(test_lungs, lab_rat, "standard gas mixture")
+	// Ensure gas alerts are accurate.
+	check_gas_alerts(lab_rat, test_lungs, test_mix)
+
+	// Test suffocation with an empty gas mix.
+	var/datum/gas_mixture/empty_test_mix = allocate(/datum/gas_mixture)
+	lab_rat = allocate(/mob/living/carbon/human/consistent)
+	test_lungs = allocate(/obj/item/organ/internal/lungs)
+	// Test one breath of nothing. Suffocate due to the breath being empty.
+	breath_ok = test_lungs.check_breath(empty_test_mix, lab_rat)
+	LUNGS_TEST_RETURN_FAIL(breath_ok, test_lungs)
+	LUNGS_TEST_SUFFOCATE(test_lungs, lab_rat, "empty gas mixture")
+	// Ensure gas alerts are accurate.
+	check_gas_alerts(lab_rat, test_lungs, empty_test_mix)
+
+	// Tests suffocation with null. This does indeed happen normally.
+	lab_rat = allocate(/mob/living/carbon/human/consistent)
+	test_lungs = allocate(/obj/item/organ/internal/lungs)
+	// Test one breath of nothing. Suffocate due to the breath being null.
+	breath_ok = test_lungs.check_breath(null, lab_rat)
+	LUNGS_TEST_RETURN_FAIL(breath_ok, test_lungs)
+	LUNGS_TEST_SUFFOCATE(test_lungs, lab_rat, "null")
+	// Ensure gas alerts are accurate.
+	check_gas_alerts(lab_rat, test_lungs, empty_test_mix)
+
+	// Tests suffocation with Nitrogen.
+	var/datum/gas_mixture/nitro_test_mix = create_nitrogen_mix()
+	lab_rat = allocate(/mob/living/carbon/human/consistent)
+	test_lungs = allocate(/obj/item/organ/internal/lungs)
+	// Test one breath of Nitrogen. Suffocate due to the breath being 100% N2.
+	breath_ok = test_lungs.check_breath(nitro_test_mix, lab_rat)
+	LUNGS_TEST_RETURN_FAIL(breath_ok, test_lungs)
+	LUNGS_TEST_SUFFOCATE(test_lungs, lab_rat, "pure nitrogen")
+	// Ensure gas alerts are accurate.
+	check_gas_alerts(lab_rat, test_lungs, nitro_test_mix)
+
+/// Tests the Plasmaman lungs organs to ensure it behaves the way we expect.
+/datum/unit_test/lungs_sanity/lungs_sanity_plasmaman/Run()
```

Comment:
I don't know why we are using a subtype if we are not going to `..()`

Reply:
If it is for the procs then I would rather the procs be global I think

Reply:
Gotcha! It was for the procs